### DISCUSSION TITLE
fuzz.ratio using levenstein distance?

### DISCUSSION THREAD

**Discussion by shubhamscifi** [rapidfuzz/RapidFuzz#208] [Q&A] [answered]
How to calculate fuzz.ratio using levenstein distance? The current implementation uses only Indel distance.

**Comment by maxbachmann** [comment, marked as answer]
You can use the [Levenshtein module](https://maxbachmann.github.io/RapidFuzz/Usage/distance/Levenshtein.html):
```python
from rapidfuzz.distance import Levenshtein

score = Levenshtein.normalized_similarity(s1, s2)
```
which returns the normalized similarity as score between 0 and 1.

**Reply by shubhamscifi** [reply]
I want to use rapidfuzz.process.extractOne also using Levenshtein distance. Which I am guessing, uses vectorized operations and hence will be faster. How can I do this?

**Reply by maxbachmann** [reply]
You can use it the same way you would with `fuzz.ratio`:
```python
from rapidfuzz.distance import Levenshtein
from rapidfuzz import process

process.extractOne(query, choices, scorer=Levenshtein.normalized_similarity)
```

**Reply by shubhamscifi** [reply]
The output of thefuzz levenshtein and rapidfuzz doesn't match even after rounding off. Why?
```
a = 'abcd'
b = 'abcde'
Indel.normalized_similarity(a,b)*100    # 88.88888888888889
Levenshtein.normalized_similarity(a,b)*100  # 80.0
thefuzz.fuzz.ratio(a,b)     # 89
```

**Reply by maxbachmann** [reply]
`fuzzywuzzy`/`thefuzz` uses the Levenshtein distance with the following weights: Insertion=1, Deletion=1, Subsitution=2. This is equivalent to the Indel distance, since any substitution can be represented by an Insertion + Deletion. You can achieve the same results using the weighted Levenshtein distance in RapidFuzz:
```
a = 'abcd'
b = 'abcde'
Levenshtein.normalized_similarity(a,b, weights=(1,1,2))*100  # 88.88888888888889
```

**Reply by shubhamscifi** [reply]
Thank you!
